src/ToastNotification/TimedToastNotification.js

```diff
@@ -0,0 +1,78 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import helpers from '../common/helpers'
+import Timer from '../common/Timer'
+import { TOAST_NOTIFICATION_TYPES } from '../common/constants'
+import { ToastNotification } from '../index.js'
+
+/**
+ * TimedToastNotification Component for Patternfly React
+ */
+class TimedToastNotification extends React.Component {
+  constructor(props) {
+    super(props)
+    helpers.bindMethods(this, ['onMouseEnter', 'onMouseLeave'])
+  }
+
+  componentDidMount() {
+    const { sticky, onDismiss, timerDelay } = this.props
+
+    if (!sticky) {
+      this.timer = new Timer(onDismiss, timerDelay)
+      this.timer.startTimer()
+    }
+  }
+
+  componentWillUnmount() {
+    this.timer && this.timer.clearTimer()
+  }
+
+  onMouseEnter() {
+    this.timer && this.timer.clearTimer()
+    const { onMouseEnter } = this.props
+    onMouseEnter && onMouseEnter()
+  }
+
+  onMouseLeave() {
+    this.timer && this.timer.startTimer()
+    const { onMouseLeave } = this.props
+    onMouseLeave && onMouseLeave()
+  }
+  render() {
+    const { children } = this.props
+    return (
+      <ToastNotification
+        {...this.props}
+        onMouseEnter={this.onMouseEnter}
+        onMouseLeave={this.onMouseLeave}
```

Comment:
The onMouseEnter and onMouseLeave functions should be run when user hovers on ToastNotificationList rather than each individual ToastNotification. User wants to pause all notifications timers when hovering over ToastNotificationList.

So to achieve this, ToastNotificationList will keep 'paused' boolean in its state and toggle it with it's onMouseEnter and onMouseLeave callbacks.

TimedToastNotification will receive 'paused' prop from ToastNotificationList via passing props to children:
```
React.Children.map(
  children,
  child => React.cloneElement(child, {
       paused: this.state.paused
  })
);
```
TimedToastNotification then updates it's timer in componentWillReceiveProps instead of doing it in onMouseEnter/Leave functions.
With this onMouseEnter/Leave props are not needed in ToastNotification unless we really want to keep it around for flexibility.

Reply:
thanks for the detailed feedback @jtomasek ... i will make this update soon :)

Reply:
@jtomasek i've made updates to account for this now...

`ToastNotificationList` now has `paused` state and propogates this to children `TimeToastNotification` (ensuring that no other toast notifications will dismiss if a single one is hovered). If any notification is hovered, all notifications will reset their timer. 

I am also allowing the consumer to observe onMouseEnter/onMouseLeave on the entire list or on each individual notification. I am aiming for as much flexibility as possible!

You can see all of the events in my updated storybook (see Action Logger while hovering):
https://rawgit.com/priley86/patternfly-react/toast-notification/index.html?knob-Label=By%20default%2C%20a%20toast%20notification%27s%20timer%20expires%20after%20eight%20seconds.&knob-Type=success&knob-Dismiss=false&knob-Menu=true&knob-Action=true&selectedKind=ToastNotification&selectedStory=Toast%20Notification%20List&full=0&down=1&left=1&panelRight=0&downPanel=storybook%2Factions%2Factions-panel